### ISSUE TITLE
Copy CSV only inserts first entry

### ISSUE THREAD

**Issue by steven-sheehy** [cockroachdb/cockroach#68484]
**Describe the problem**

Attempting to port my company's application from PostgreSQL to CockroachDB for a PoC. Our application uses `COPY <table> FROM STDIN WITH CSV` to ingest our bulk data at high insertion rates. The same code and data that works with PostgreSQL does not work with CockroachDB. It inserts only the first entry in the CSV data and ignores the rest of the lines. No error occurs, just missing data.

If I switch from `CSV` to `TEXT` format with comma delimiter it inserts all rows successfully. Though we are unable to use `TEXT` mode for our real application due to its limitations with encoding more complicated data.

If I use psql to attempt to reproduce the problem, I don't encounter the issue. It seems to be due specifically CockroachDB's interaction with JDBC when using `CopyManager`.

**To Reproduce**

I've created a simplified test project that reproduces the behavior. The project uses the in-memory cockroach `movr` database and first inserts data using `TEXT` successfully. Truncates the data then inserts the same data in CSV mode and fails.

1. `cockroach demo`
2. `git clone https://github.com/steven-sheehy/cockroachdb-copy.git`
3. `cd cockroachdb-copy`
4. `./mvnw spring-boot:run -Dspring-boot.run.arguments=--spring.datasource.password=<password>`
5. See assertion error

**Expected behavior**
All rows to be inserted

**Additional data / screenshots**
`COPY users FROM STDIN CSV;`
```
8a3d70a3-d70a-4000-8000-00000000001d,denver,Hannah,400 Broad St,0987654321
9eb851eb-851e-4800-8000-00000000001e,denver,Carl,53 W 23rd St,5678901234
```

**Environment:**
 - CockroachDB version v21.1.6
 - Server OS: MacOS
 - Client app: JDBC

**Additional context**
What was the impact?

Unable to move forward with the PoC.

**Comment by RichardJCai** [comment]
cc @cockroachdb/bulk-io

**Comment by rafiss** [comment]
Thanks for filing this @steven-sheehy! Your repro code is very helpful. I looked into this and it seems to be a bug in CockroachDB.

We'll work on a fix, but in the meantime, have you checked if the `IMPORT INTO ... CSV` command unblocks you?

https://www.cockroachlabs.com/docs/stable/import-into.html

**Comment by steven-sheehy** [comment]
Thanks @rafiss! Unfortunately uploading our files to cloud storage would negate any performance benefit of using `COPY` since the files are batched every two seconds. I can see that being useful for hourly or daily ETL processes.

Let us know if the fix becomes available in any pre-release docker image and we can test it out.

**Comment by rafiss** [comment]
@steven-sheehy sure thing - the fix will be in the first alpha release of 21.2 and will also be backported to 21.1. both are scheduled to be released around the end of August
